Enable c++11

davidjmott opened the pull request.

Commit:
Enable c++11

drrtuy commented on CMakeLists.txt:
```diff
@@ -174,7 +163,7 @@ SET (ENGINE_LOCALDIR   "${INSTALL_ENGINE}/local")
 SET (ENGINE_MYSQLDIR   "${INSTALL_ENGINE}/mysql")
 SET (ENGINE_TOOLSDIR   "${INSTALL_ENGINE}/tools")
 
-SET (ENGINE_COMMON_LIBS     messageqcpp loggingcpp configcpp idbboot ${Boost_LIBRARIES} xml2 pthread rt libmysql_client ${JEMALLOC_LIBRARIES})
```
Are you sure about that? We still didn't provide a substituion.

drrtuy commented on CMakeLists.txt:
```diff
@@ -83,14 +91,6 @@ INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/build/releasenum DESTINATION ${INSTALL
-exec_program("git"
-    ${CMAKE_CURRENT_SOURCE_DIR}
-    ARGS "describe --match=NeVeRmAtCh --always --dirty"
-    OUTPUT_VARIABLE GIT_VERSION)
```
It seems the removal of this part causes BB to fail when it tries to copy gitVersion.

Commit:
rollback unintended changes

Commit:
Fully qualify ambiguous isnan() with std::

Commit:
fix ambiguous symbol

Commit:
move cmake scripts to cmake folder
add boost super build project (currently disabled)
declare BOOST_NO_CXX11_SCOPED_ENUMS on projects that use boost::filesystem

drrtuy reviewed (approved):
Mcol1408 error isn't related with the change.
